### PR TITLE
Add additional type conversions

### LINKED ISSUE #33: unsigned value support
Hi,

is there any specific reason we dont support unsigned value conversion?

### PR DESCRIPTION
**Pull request by moorereason**
Adds float32, uint, uint64, uint32, uint16, and uint8 throughout.

Fixes #33

### CODE REVIEW

File: cast_test.go [hunk 1]
```diff
@@ -14,6 +14,231 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestToUintE(t *testing.T) {
+	tests := []struct {
+		input  interface{}
+		expect uint
+		iserr  bool
+	}{
+		{int(8), 8, false},
```
Comment (bep) [inline cast_test.go]: Uint tests without any negative values ... I guess that should have returned an error.

Comment (moorereason) [inline cast_test.go]: Well, duh.  Maybe I should slow down a little.  :roll_eyes: 

New commit added.

File: caste.go [hunk 2]
```diff
@@ -612,14 +696,29 @@ func ToUint8E(i interface{}) (uint8, error) {
 		}
 		return 0, fmt.Errorf("unable to cast %#v to uint8: %s", i, err)
 	case int:
+		if s < 0 {
+			return 0, fmt.Errorf("unable to cast negative value to uint8")
```
Comment (bep) [inline caste.go]: I understand some of the code duplication here, but the error could be created once and reused.

Comment (moorereason) [inline caste.go]: I was trying to avoid allocating the error unless I needed it, but I can refactor if you want.  Let me know what you want to do.

Comment (moorereason) [inline caste.go]: How about this?

```go
func errorNegativeUint(size string) error {
    return errors.New("unable to cast negative value to uint" + size)
}

if s < 0 {
    return 0, errorNegativeUint("8")
}
```

Or am I over-engineering this?  :smile:

Comment (bep) [inline caste.go]: You are over-engineering slightly. People know what func they called, so you could just add a private member err variable at the top of the file with a more generic message, `errNegativeNotAllowed = "unable to cast negative value"`, and then return that.

Comment (moorereason) [inline caste.go]: Pushed another commit.  If you like it, I can squash the commits.

**Pull request merged**